cmd/net-rpc-client.go

```diff
@@ -150,8 +150,18 @@ func (rpcClient *RPCClient) Call(serviceMethod string, args interface{}, reply i
 			// Close the underlying connection.
 			rpcLocalStack.Close()
 
-			// Set rpc error as rpc.ErrShutdown type.
-			err = rpc.ErrShutdown
+			// Try once more to re-connect.
+			rpcLocalStack, err = rpcClient.dialRPCClient()
+			if err != nil {
+				return err
+			}
+
+			// Attempt the service call oncne again, upon any error now just give up.
+			err = rpcLocalStack.Call(serviceMethod, args, reply)
+			if err != nil && err.Error() == rpc.ErrShutdown.Error() {
+				// Set rpc error as rpc.ErrShutdown type.
+				err = rpc.ErrShutdown
```

Comment:
no we need to set it back to a proper type so that the type is not lost. otherwise rpc.Call() converts this into rpc.ServerError() at times not all the time.